### ISSUE TITLE
WithinPosition as a valid feature location

### ISSUE THREAD

**Issue by JamesJeffryes** [biopython/biopython#1490]
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

2.7.10 (default, Feb  7 2017, 00:08:15) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
CPython
Darwin-16.7.0-x86_64-i386-64bit
1.70

### Expected behaviour
I expect to be able to specify a "WithinPosition" as in a genbank file in accordance with http://www.insdc.org/files/feature_table.html#3.4.3

> 102.110                   Indicates that the exact location is unknown but that it is 
>                           one of the bases between bases 102 and 110, inclusive

I'd expect an output similar to if a single base is specified:

`SeqFeature(FeatureLocation(WithinPosition(101, left=101, right=109), WithinPosition(101, left=101, right=109), strand=1), type='CDS')`

### Actual behaviour

It appears that the current implantation only accepts a "Within Position" as one part of a feature location (with parentheses) rather than accepting it as an location itself.

I get the following error:
```
/Library/Python/2.7/site-packages/Bio/GenBank/__init__.py:1143: BiopythonParserWarning: Couldn't parse feature location: '102.110'
  % location_line))
```

### Steps to reproduce

Here's a truncated genbank file that should cause the error on parsing:
[minimal.txt](https://github.com/biopython/biopython/files/1604128/minimal.txt)

If you think the behavior I proposed is reasonable, I may be able to submit a PR to implement the change

**Comment by peterjc** [comment]
Problem entry:

```
LOCUS       SCU49845     5028 bp    DNA             PLN       21-JUN-1999
DEFINITION  Saccharomyces cerevisiae TCP1-beta gene, partial cds, and Axl2p
            (AXL2) and Rev7p (REV7) genes, complete cds.
ACCESSION   U49845
VERSION     U49845.1  GI:1293613
KEYWORDS    .
SOURCE      Saccharomyces cerevisiae (baker's yeast)
  ORGANISM  Saccharomyces cerevisiae
...
     CDS             102.110
                     /codon_start=3
                     /product="TCP1-beta"
                     /protein_id="AAA98665.1"
                     /db_xref="GI:1293614"
                     /translation="SSIYNGISTSGLDLNNGTIADMRQLGIVESYKLKRAVVSSASEA
                     AEVLLRVDNIIRARPRTANRQHM"
...
```

Where did you get the file from? https://www.ncbi.nlm.nih.gov/nuccore/U49845 does not have this feature, the annotation there is rather different.

What are you suggesting as the parser location structure?

**Comment by JamesJeffryes** [comment]
A colleague of mine made this file to test a genome parser I am working on
that uses biopython by mutating an existing genbank file. As such the
feature does not exist in the source and is nonsensical (a point feature
would not the other qualifiers) but the syntax is valid as far as I can
tell.

I'd recommend using the same WithinPosition as the start and end of the
feature location similar to how a single base location is represented.

**Comment by peterjc** [comment]
I'm not convinced that representation makes sense... can you find a real world GenBank file from the NCBI which might help me?

(BTW, I'm not saying testing the parser with mutated genbank file is a bad idea - its a very good idea, especially if it catches things like infinite loops etc. But here we have a very clear warning, and one problematic feature has its location ignored which is a graceful way to handle this and a big improvement over past versions of Biopython which would have aborted with an exception)

**Comment by jkbaumohl** [comment]
Peter please look at http://www.insdc.org/files/feature_table.html#3.4.3
It is pretty clear that "123.235" is a valid position information. 
Additionally I have contacted RefSeq : 
Here is that inquiry and answer:
Comment: Hello I am working on GenBank uploader into KBase. We are using the biopython parser but ran into an interesting discrepancy in a unit testing for "non-standard" locations.

So based on the documentation I found
http://www.insdc.org/files/feature_table.html#3.4.3

The single dot notation is for :
102.110 Indicates that the exact location is unknown but that it is
one of the bases between bases 102 and 110, inclusive


However, the biopython parser chokes on the test file that has that usage:
misc_feature 9938.9949
/gene="SomewhereInRange"
/locus_tag="SomewhereInRange"

When digging into the biopython code it appears they expect a different way of indicating that. From biopython:
_within_position = r"\(\d+\.\d+\)"

_re_within_position = re.compile(_within_position)

_within_location = r"([<>]?\d+|%s)\.\.([<>]?\d+|%s)" \

% (_within_position, _within_position)

assert _re_within_position.match("(3.9)")

assert re.compile(_within_location).match("(3.9)..10")

assert re.compile(_within_location).match("26..(30.33)")

assert re.compile(_within_location).match("(13.19)..(20.28)")

So I am wondering first off, which of these notations is correct? If it turns out the documentation is correct, but biopython parser is wrong, I will submit a ticket to biopython.

Thank you,
Jason Baumohl
--------------
Their Response:
Hi Jason,

It looks like there is a problem with the BioPython logic.The parentheses are NOT required or expected for this example, which would denote a single base somewhere in the bp 3-9 range:
assert _re_within_position.match("(3.9)")

But the parentheses are required for these examples, for a range with one or both ends that aren't precisely defined:
assert re.compile(_within_location).match("(3.9)..10")
assert re.compile(_within_location).match("26..(30.33)")
assert re.compile(_within_location).match("(13.19)..(20.28)")

That said, we suspect that there is no current data in GenBank with such locations.We haven't done a comprehensive scan, so I can't guarantee that there aren't some examples, but it was never supported in Sequin, and I think not in tbl2asn, so it's highly unlikely that anyone was able to construct such data. So this would be really, really low priority to fix in the BioPython parser.

Best regards,
-Terence

-Terence Murphy, Ph.D.
Staff Scientist
RefSeq and Gene Team Lead
NCBI/NLM/NIH/DHHS

**Comment by peterjc** [comment]
I agree that as per  http://www.insdc.org/files/feature_table.html#3.4.3 a location ``123.456`` is currently considered valid GenBank syntax,

```
102.110                   Indicates that the exact location is unknown but that it is 
                          one of the bases between bases 102 and 110, inclusive
```

But I don't recall ever seeing one, and don't think the current Biopython object model can adequately express this.

And I think Terence is right, that regular expression you pointed to handles horrible locations like:

```
assert re.compile(_within_location).match("(3.9)..10")
assert re.compile(_within_location).match("26..(30.33)")
assert re.compile(_within_location).match("(13.19)..(20.28)")
```

It is interesting that he suspects there are no current examples like ``123.456`` in GenBank, which fits with the fact that no-one has run into this until you (with a fuzzed example file).

Would a more specific warning be an acceptable compromise, rather than the generic one we give now:

```
BiopythonParserWarning: Couldn't parse feature location: '102.110'
```
